### PR TITLE
Add VDPA block device simulator utils

### PR DESCRIPTION
**Pull request by qingwangrh**
ID:2177943

Signed-off-by: qingwangrh <qinwang@redhat.com>

### CODE REVIEW

**Comment by qingwangrh** [comment]
passed test code like :
test = VDPABlkSimulatorTest(test)
    test.setup()
    name0 = "blk0"
    name1 = "blk1"
    a = test.add_vdpa_blk_dev(name0)
    print(a)
    b = test.add_vdpa_blk_dev(name1)
    print(b)

    test.remove_vdpa_blk_dev(name1)
    test.remove_vdpa_blk_dev(name0)
    test.cleanup()

**Comment by XueqiangWei** [comment]
> ID:2177943

No "signed-off-by: ***" message in it.

File: provider/vdpa_sim_blk_utils.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+from virttest.utils_kernel_module import KernelModuleHandler
+
+LOG = logging.getLogger('avocado.test')
+
+
+class VDPABlkSimulatorTest(object):
+
+    def __init__(self, test):
+        self._test = test
+        pass
```
Comment (XueqiangWei) [inline provider/vdpa_sim_blk_utils.py:16]: I think we can remove it.

Comment (qingwangrh) [reply provider/vdpa_sim_blk_utils.py]: updated

File: provider/vdpa_sim_blk_utils.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+LOG = logging.getLogger('avocado.test')
+
+
+class VDPABlkSimulatorError(Exception):
+    """ General VDPA BLK error"""
+    pass
+
+
+class VDPABlkSimulatorTest(object):
```
Comment (YongxueHong) [inline provider/vdpa_sim_blk_utils.py:17]: Hi @qingwangrh 
I think we can inherit the `VDPASimulatorTest` https://github.com/avocado-framework/avocado-vt/blob/49e33beadd26651f184ab7ad4ae114c3a1320cf5/virttest/utils_vdpa.py#L205
then override the `setup` method and add the `add_vdpa_blk_dev`, `remove_vdpa_blk_dev` methods according to the `blk simulator` requirements.
CC @XueqiangWei

**Review by YongxueHong — DISMISSED** [review]
LGTM

**Comment by XueqiangWei** [comment]
LGTM,
Acked-by: xuwei@redhat.com

File: provider/vdpa_sim_blk_utils.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        process.run(cmd, shell=True)
+        time.sleep(2)
+        disks = process.system_output(disk_cmd, shell=True).decode().split()
+        dev_after = set(disks)
+        host_dev = list(dev_after - dev_before)
+        if not host_dev:
+            raise VDPABlkSimulatorError("vdpa dev add %s failed" % name)
+        return host_dev[0]
+
+    def remove_vdpa_blk_dev(self, name):
```
Comment (YongxueHong) [inline provider/vdpa_sim_blk_utils.py:66]: Please add the docstring for this method.

**Review by YongxueHong — APPROVED** [review]
LGTM.

**Pull request merged**